File: Solution-Go/Longest_424.go
func characterReplacement(s string, k int) int {
    l, maxCount := 0, 0
    freq := make(map[rune]int)
    for r, c := range s {
        freq[c] += 1
        maxCount = max(maxCount, freq[c])
        if r - l + 1 > maxCount + k {
            freq[rune(s[l])] -= 1
            l += 1
        }
    }
    return len(s) - l
}

func max(x int, y int) int {
    if x > y {
        return x
    }
    return y
}

File: Solution-Java/Longest_424.java
class Longest_424 {
    public int characterReplacement(String s, int k) {
        int[] freq = new int[26];
        int l = 0;
        int maxCount = 0;
        for (int r = 0; r < s.length(); r++) {
            maxCount = Math.max(maxCount, ++freq[s.charAt(r) - 'A']);
            if (r - l + 1 > maxCount + k) --freq[s.charAt(l++) - 'A'];
        }
        return s.length() - l;
    }
}